### ISSUE TITLE
GTAV now at version 1.0.1180.1 ScriptHookV broken

### ISSUE THREAD

**Issue by IanKirwan** [ai-tor/DeepGTAV#67]
Hi,
The latest GTAV update has broken ScriptHookV, so will have to wait for a new release of ScriptHookV. I suspect this will also break many , if not all, off the native offsets, rendering DeepGTAV in operable too.

I have asked Rockstar if there is any way to freeze GTAV at a version and even work offline continuously so we dont have to go through this every time they do an update.

Is anyone aware of any work arounds?

Many thanks

**Comment by ai-tor** [comment]
I don't know of any solution except keeping a compressed copy of your GTAV installation dir and replace your GTAV installation with it everytime Rockstar releases an update, and then start GTAV without connection to the internet, to avoid the game from updating again.

**Comment by IanKirwan** [comment]
Hi @ai-tor,

It forces me to connect. Will only let me use the saved game once before demanding connection to the Social Club. Do you have a work around of that?

Cheers
Ian

**Comment by dominikandreas** [comment]
The launcher bypass mod has worked well for me:
https://de.gta5-mods.com/tools/gtavlauncherbypass

It's also a real timesaver for debugging since it leads to a much faster
startup

**Comment by ai-tor** [comment]
Holy smokes! Didn't know about it!

**Comment by IanKirwan** [comment]
Just found that one too, but nice one Dominik. I'll still have to wait for the ScriptHookV update now GTAV is updated, but at least I can freeze this version :)

**Comment by szpon** [comment]
Hi,
ScriptHookV was updated and now supports GTAV v1180.1, but I'm getting `FATAL: Can't find native` error as soon as I run VPilot.

**Comment by IanKirwan** [comment]
Sounds like DEEP GTAV is not running. Check the DEEP GTAV log file

**Comment by szpon** [comment]
It seems to be loading but crashing just after VPilot make connection.

DeepGTAV log:
```
Initializing Winsock...Initialized.
Socket created.
Bind done.
Listening...
Connection accepted.
Start received
```

VPilot:
```
➜  VPilot git:(master) ✗ ./dataset --host=192.168.2.35
Trying to connect to DeepGTAV
Successfully connected to DeepGTAV
Traceback (most recent call last):
  File "./dataset", line 46, in <module>
    message = client.recvMessage()
  File "/Users/marcinzadka/Desktop/DeepGTAV/VPilot/deepgtav/client.py", line 50, in recvMessage
    frame = self._recvall()
  File "/Users/marcinzadka/Desktop/DeepGTAV/VPilot/deepgtav/client.py", line 64, in _recvall
    packet = self.s.recv(4 - len(data))
ConnectionResetError: [Errno 54] Connection reset by peer
```

But i had issues (#70) with DeepGTAV and older GTA V version, so this may be somehow related.

**Comment by IanKirwan** [comment]
Did you rebuild DeepGTAV with the new ScriptHookV SDK?
I haven't got past that yet. The build has errors:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	E0167	argument of type "Vector3 *" is incompatible with parameter of type "Any *"	DeepGTAV	g:\Git\DeepGTAV\Scenario.cpp	339	
Error (active)	E0167	argument of type "Vector3 *" is incompatible with parameter of type "Any *"	DeepGTAV	g:\Git\DeepGTAV\Scenario.cpp	339	
Error (active)	E0167	argument of type "Vector3 *" is incompatible with parameter of type "Any *"	DeepGTAV	g:\Git\DeepGTAV\Scenario.cpp	476	
Error (active)	E0167	argument of type "Vector3 *" is incompatible with parameter of type "Any *"	DeepGTAV	g:\Git\DeepGTAV\Scenario.cpp	476	
Error (active)	E0167	argument of type "int *" is incompatible with parameter of type "Any"	DeepGTAV	g:\Git\DeepGTAV\Scenario.cpp	617	
Error (active)	E0167	argument of type "float *" is incompatible with parameter of type "Any"	DeepGTAV	g:\Git\DeepGTAV\Scenario.cpp	617	
Error	C2664	'void ENTITY::GET_ENTITY_MATRIX(Entity,Any *,Any *,Vector3 *,Vector3 *)': cannot convert argument 2 from 'Vector3 *' to 'Any *'	DeepGTAV	G:\Git\DeepGTAV\Scenario.cpp	339	
Error	C2664	'void ENTITY::GET_ENTITY_MATRIX(Entity,Any *,Any *,Vector3 *,Vector3 *)': cannot convert argument 2 from 'Vector3 *' to 'Any *'	DeepGTAV	G:\Git\DeepGTAV\Scenario.cpp	476	
Error	C2664	'Any PATHFIND::GENERATE_DIRECTIONS_TO_COORD(float,float,float,Any,Any,Vehicle *,Any)': cannot convert argument 5 from 'int *' to 'Any'	DeepGTAV	G:\Git\DeepGTAV\Scenario.cpp	617	
```

**Comment by IanKirwan** [comment]
I recast the params identified in the error list as (Any*) or (Any) where appropriate but I'm getting the same problem. ScripHookV.log:

```
[11:00:06] INIT: Started
[11:00:06] INIT: Success, game version is VER_1_0_1180_2_NOSTEAM
[11:00:06] INIT: Registering script 'DeepGTAV.asi' (0x00007FFF9F16EAF0)
[11:00:11] INIT: Pool 1 extended
[11:00:11] INIT: Pool 2 extended
[11:00:11] INIT: Pool 3 extended
[11:00:11] INIT: Pool 4 extended
[11:00:16] INIT: GtaThread collection size 189
[11:00:16] INIT: wnd proc 0x00000000FFFF048F
[11:00:16] INIT: IDXGISwapChain 0x00007FFFB6F9EF40 (0x00007FFFB6F51DB0)
[11:00:16] INIT: IDXGISwapChain set
[11:00:16] INIT: DX init started
[11:00:16] INIT: DX feature level B000
[11:00:16] INIT: DX init succeeded
[11:00:58] CORE: Requesting thread creation
[11:00:58] CORE: Creating threads
[11:00:59] CORE: Started control thread, id 31 active 1
[11:00:59] CORE: Started thread 'DeepGTAV.asi' (0x00007FFF9F16EAF0), id 32 active 1
[11:00:59] CORE: Launching main() for 'DeepGTAV.asi' (0x00007FFF9F16EAF0), id 32
[11:01:44] FATAL: Can't find native 0x80E4A6EDDB0BE8D9
[11:02:38] UNINIT: Unregistering script 'DeepGTAV.asi'
```

**Comment by ai-tor** [comment]
Hi all, I will update DeepGTAV in a while, but to fix the issue for now, just modify the declarations in ```lib/natives.h``` to match the argument types indicated by the compiler. That should solve the compilation issues. For example:

```GET_ENTITY_MATRIX(Entity entity, Any* p1, Any* p2, Any* p3, Any* p4)``` -> ```GET_ENTITY_MATRIX(Entity entity, Vector3* p1, Vector3* p2, Vector3* p3, Vector3* p4)```

This happens because I've figured out some of the argument types myself, but those changes are not merged with the official ScriptHookV release, so each time there is an update I have to set them up again. Now, with so many users, you are usually faster than me when running the new releases :)

**Comment by IanKirwan** [comment]
Thanks @ai-tor.
OK will try this instead of casting the params in the calls.

**Comment by IanKirwan** [comment]
What to do about?:```

Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	E0167	argument of type "int *" is incompatible with parameter of type "Any"	DeepGTAV	g:\Git\DeepGTAV\Scenario.cpp	617	
Error (active)	E0167	argument of type "float *" is incompatible with parameter of type "Any"	DeepGTAV	g:\Git\DeepGTAV\Scenario.cpp	617	
```
Suggested changes:
In **natives.h**
```
GENERATE_DIRECTIONS_TO_COORD(float x, float y, float z, Any p3, Any p4, Vehicle* vehicle, Any p6) ->
GENERATE_DIRECTIONS_TO_COORD(float x, float y, float z, Any p3, Any p4, Vehicle* vehicle, Any p6)
```

In **Scenario.cpp**
```
GENERATE_DIRECTIONS_TO_COORD(dir.x, dir.y, dir.z, TRUE, &direction, &vehicle, &distance);
GENERATE_DIRECTIONS_TO_COORD(dir.x, dir.y, dir.z, TRUE, direction, &vehicle, distance);
```

**Comment by ai-tor** [comment]
Probably this: ```GENERATE_DIRECTIONS_TO_COORD(float x, float y, float z, Any p3, int* p4, Vehicle* vehicle, float* p6)```

**Comment by IanKirwan** [comment]
OK, will try that now. My suggestion still crashes GTAV.

**Comment by ai-tor** [comment]
In scenario.cpp it should be ```GENERATE_DIRECTIONS_TO_COORD(dir.x, dir.y, dir.z, TRUE, &direction, &vehicle, &distance)```

**Comment by IanKirwan** [comment]
Yep, that makes more sense than what i did. GTAV loading will have an asnwer in a few mins.

**Comment by IanKirwan** [comment]
Nay, there are other gremlins at work it seems:
```
[11:45:01] CORE: Started thread 'DeepGTAV.asi' (0x00007FFF9F16EA70), id 32 active 1
[11:45:01] CORE: Launching main() for 'DeepGTAV.asi' (0x00007FFF9F16EA70), id 32
[11:45:30] FATAL: Can't find native 0x80E4A6EDDB0BE8D9
[11:46:13] UNINIT: Unregistering script 'DeepGTAV.asi'
```
Will have a quick look at this 0x80E4A6EDDB0BE8D9 reference to see if there is anything obvious.

**Comment by IanKirwan** [comment]
Its:
`LOAD_ALL_PATH_NODES(BOOL p0) { return invoke<BOOL>(0x80E4A6EDDB0BE8D9, p0)`

**Comment by szpon** [comment]
@IanKirwan [This](https://github.com/ai-tor/DeepGTAV/issues/71#issuecomment-327096817) might be helpful

**Comment by IanKirwan** [comment]
Thanks @szpon
i got some catch up to do.

**Comment by ai-tor** [comment]
Fixed, as commented in #71

**Comment by j-lnaddef** [comment]
I am getting the same issue now that the game is in version 1.0.1290.1, and I did not know about the launcher bypass...
Any new release planned? @ai-tor
